### DISCUSSION TITLE
Camera should follow animation.

### DISCUSSION THREAD

**Discussion by atul161** [google/filament#6469] [Q&A] [answered]
First of all, thank you to everyone who is working on this project. 

I have a use case where there is an animation which is attached to the model.
Animation has translation rotation also. 

That's the region sometime my model goes out of my screen.

Is there any way through which we can move the camera according to the animation so that it will not go offscreen?

It will be very helpful if there is any API through which we can do this.

Thanks!

**Comment by romainguy** [comment, marked as answer]
You can do this by animating the camera itself so its "look at" position always points at your object.

**Reply by atul161** [reply]
Thankyou! @romainguy ! For all the help!

**Reply by atul161** [reply]
@romainguy my object has animation also(Translation) . So is there any way to get the current target position? 
I have tried with getworldTransform...but it is returning 0 always.

**Reply by atul161** [reply]
I think there would be an API through which we can attach the camera to the model, irrespective of the animation.

Currently in the camera manipulator, if I  set the target position then it will be fixed for whole object irrespective of animation.

I have also attached a camera look at inside the render method. It is working first time but not the second time. I am calling transform manager and getting the world position and setting it into eye look at. It leads to crashes sometimes.

Also, I think some objects are not destroyed properly when I move back from an activity. 

Because when I go a second time then the entity count is 2x the previous one.

I have also called destroy model still it doesn't work.

**Reply by atul161** [reply]
Get world transform giving incorrect value. The first time when I load the model then everything worked. But from the second time, world transform gives incorrect value for the entity which have animations.
I am calling getFirstentitybyname.

**Reply by atul161** [reply]
for(entity in entities){
            if(assetForModel!!.getName(entity) == "Hips"){
                hip = entity
                break
            }
        }
        cameraManipulatorForModel.getLookAt(eyePos, target, upward)
        val hipsMovement: DoubleArray =
            engine.transformManager.getWorldTransform(hip, null as DoubleArray?)
        Log.d("Hips","X"+hipsMovement[12] + "Y" + hipsMovement[13].toString() + "Z" +hipsMovement[14])

        Log.d("Hip",hip.toString())
        Log.d("Eye","X"+eyePos[0] + "Y" + eyePos[1].toString() + "Z" + eyePos[2])
        camera.lookAt(
            eyePos[0],eyePos[1],hipsMovement[14] + deltaz,
            hipsMovement[12],hipsMovement[13],hipsMovement[14],
            upward[0], upward[1], upward[2]
        )



Here is code that i have tried. when i load model first time then my camera is moviing with the hip. In that case i am also getting world transform co ordinate. But when i come again to the same activity then the hip entity count is different and world transform is 0. Can you please help me out.
